benchmark/synthetic/models/lstm_text_generation_wikitext2.py

```diff
@@ -0,0 +1,190 @@
+                          optimizer=optimizer,
+                          metrics=['accuracy'],
+                          target_tensors=[targets])
+        else:
+            model.compile(loss='categorical_crossentropy', optimizer=optimizer)
+
+        # create a distributed trainer for cntk
+        if keras.backend.backend() == "cntk" and gpus > 1:
+            start, end = cntk_gpu_mode_config(model.model, x_train.shape[0])
+            x_train = x_train[start: end]
+            y_train = y_train[start: end]
+
+        time_callback = timehistory.TimeHistory()
+
+        def sample(preds, temperature=1.0):
+            # helper function to sample an index from a probability array
+            preds = np.asarray(preds).astype('float64')
+            preds = np.log(preds) / temperature
+            exp_preds = np.exp(preds)
+            preds = exp_preds / np.sum(exp_preds)
+            probas = np.random.multinomial(1, preds, 1)
+            return np.argmax(probas)
+
+        def on_epoch_end(epoch, logs):
+            # Function invoked at end of each epoch. Prints generated text.
+            print()
+            print('----- Generating text after Epoch: %d' % epoch)
+
+            start_index = random.randint(0, len(text) - maxlen - 1)
+            for diversity in [0.2, 0.5, 1.0, 1.2]:
+                print('----- diversity:', diversity)
+
+                generated = ''
+                sentence = text[start_index: start_index + maxlen]
+                generated += sentence
+                print('----- Generating with seed: "' + sentence + '"')
+                sys.stdout.write(generated)
+
+                for i in range(400):
+                    x_pred = np.zeros((32, maxlen, len(chars)))
+                    for t, char in enumerate(sentence):
+                        x_pred[0, t, char_indices[char]] = 1.
+
+                    preds = model.predict(x_pred, verbose=0)[0]
+                    next_index = sample(preds, diversity)
+                    next_char = indices_char[next_index]
+
+                    generated += next_char
+                    sentence = sentence[1:] + next_char
+
+                    sys.stdout.write(next_char)
+                    sys.stdout.flush()
+                print()
+
+        print_callback = LambdaCallback(on_epoch_end=on_epoch_end)
+
+        if inference:
+            callback = print_callback
+        else:
+            callback = time_callback
+
+        if use_dataset_tensors:
+            model.fit(epochs=self.epochs, steps_per_epoch=15,
+                      callbacks=[callback])
+        else:
+            model.fit(x_train, y_train,
+                      batch_size=self.batch_size,
+                      epochs=self.epochs,
+                      callbacks=[callback])
+
+        if keras.backend.backend() == "tensorflow":
+            keras.backend.clear_session()
+
+        if keras.backend.backend() == "cntk" and gpus > 1:
```

Comment:
remove

Reply:
Done. Added a check for MXNet and TensorFlow backend. Now, the benchmark script only supports MXNet and TensorFlow as backend and fails for rest.
Thanks.